File: test/files/jvm/nest.java
package a;


/** This file is needed for test 'nest.scala'. It should
 *  be compiled with javac and packaged into lib/nest.jar
 */
public class nest {
    public static class best {
        public static class rest {
            public static rest test = new rest();
            public static int  x = 10;
            public int inc(int i) {
                return i + 1;
            }
        }
    }
}

File: test/files/jvm/nest.scala
//############################################################################
// Test Scala interaction with Java nested classes and static members.
//############################################################################

/** found in nest.jar, compiled from nest.java */
import nestpkg._;

object Test extends App {
  val x = nest.best.rest.test
  Console.println(x.inc(1))

  val o = new nest.best;
  val r = new nest.best.rest;
  Console.println(nest.best.rest.test.inc(2))
  Console.println(nest.best.rest.x)

  print("Instantiating public inner class: ")
  val outer = new nest
  val inn   = new outer.Inn(42)
  inn.doSomething
}
